### PR TITLE
Added new models and views

### PR DESCRIPTION
**Pull request by vipulyadav150**
Added Company, Transaction, InvestmentRecords and CompanyCMPRecords models along with ProfileView for viewing company profiles along with the transaction handling model , so that once a user selects a company of his/her choice and is directed to the profile page of company he can make transactions in two modes (buy or sell) and the transaction in any of these modes is handled in Transaction Handling View (CompanyTransactionView) where the transaction can either take place or not depending factors it depends on (i.e it depends on various attributes like the company stocks remaining / offered, user cash, etc...) and all the depending models such as InvestmentRecord and Company model attribute values are also updated.

Each user transaction creates a new row in Transaction table with updated values such as user_net_worth and all related models have their attributes values updated such as stocks of each company owned by user etc

### CODE REVIEW

**Review by shan18 — CHANGES_REQUESTED** [review]
Were the unnecessary files and code lines removed?

File: accounts/models.py [hunk 1]
```diff
@@ -1,6 +1,10 @@
 from django.conf import settings
 from django.db import models
 from django.contrib.auth.models import AbstractBaseUser, BaseUserManager
+from decimal import Decimal
+import numpy as np
+
+DEFAULT_LOAN_AMOUNT = getattr(settings, 'DEFAULT_LOAN_AMOUNT', Decimal(12000.00))
```
Comment (shan18) [inline accounts/models.py:7]: The bank model hasn't been created yet so is this line of code necessary?

File: accounts/models.py [hunk 2]
```diff
@@ -85,3 +91,19 @@ def has_module_perms(self, app_label):
...
+            self.cash -= Decimal(quantity) * price
+            self.save()
+            return True
+        return False
+
+    def sell_stocks(self, quantity, price):
+        self.cash += Decimal(quantity) * price
+        self.save()
+
+    def update_cv(self, net_worth_list):
```
Comment (shan18) [inline accounts/models.py:108]: This leaderboard ranking method may change in future so add a comment on top of this function saying TODO: this needs to be changed.

File: accounts/views.py [hunk 3]
```diff
@@ -37,3 +38,8 @@ def form_valid(self, form):
         super(RegisterView, self).form_valid(form)
         messages.success(self.request, 'Verification link sent! Please check your email.')
         return redirect(self.success_url)
+
```
Comment (shan18) [inline accounts/views.py]: Remove these empty blank lines at the end of the file. According to PEP8, each file should have just a single blank line at the end.

File: market/forms.py [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+class StockTransactionForm(forms.Form):
+    mode = forms.ChoiceField(choices=TRANSACTION_MODES)
+    quantity = forms.CharField(required=True,
+                               widget=forms.TextInput(attrs={
+                                   'class': 'form-control',
+                                   'placeholder': 'Enter integers only',
+                                   'title': 'Enter integers only',
+                                   'pattern': '[0-9]+'
+                               }))
```
Comment (shan18) [inline market/forms.py]: The quantity variable line should be written in the following way according to PEP 8 guidelines.

```[python]
quantity = forms.CharField(
    required=True,
    widget=forms.TextInput(attrs={
        'class': 'form-control',
        'placeholder': 'Enter integers only',
        'title': 'Enter integers only',
        'pattern': '[0-9]+'
    })
)
```

File: market/models.py [hunk 5]
```diff
@@ -0,0 +1,240 @@
...
+    def get_by_company(self, company):
+        return self.get_queryset().get_by_company(company=company)
+
+    def calculate_net_worth(self, user):
+        qs = self.get_by_user(user)
+        amount = Decimal(0.00)
+        for inv in qs:
+            amount += Decimal(inv.stocks) * inv.company.cmp
+        return amount + user.cash
+
```
Comment (shan18) [inline market/models.py:191]: PEP8: Only two blank lines should be present between any two classes.

**Review by shan18 — CHANGES_REQUESTED** [review]
Please go through the comments mentioned in this review.

**Comment by shan18** [comment]
Take care of this in all the files:
each file should end with exactly one blank line (PEP8)

File: accounts/models.py (hunk 1 above)
Comment (vipulyadav150) [reply accounts/models.py:7]: Yes, it is required in User Model

**Review by shan18 — APPROVED** [review]

**Pull request merged**